tests/kola/misc-ro/misc-ro.sh

```diff
@@ -137,3 +137,8 @@ if [[ ! $(rpm -q NetworkManager) =~ 'rhaos4.7' ]]; then
     fatal "NetworkManager package changed from rhaos4.7 branch. Needs investigation."
 fi
 echo "ok NetworkManager package comes from rhaos4.7 branch."
+
+# Ensure NM's internal DHCP client runs by default
+if ! journalctl -b 0 -u NetworkManager --grep="dhcp-init: Using DHCP client 'internal'"; then
```

Comment:
One thing I noticed; `journalctl --grep` still exits with `0` even if there are no matches.  So perhaps e.g.:
```
$ if ! journalctl -b 0 -u NetworkManager --grep=dhcp | grep -q "Using DHCP client 'internal'"; then
```
?

Reply:
I wouldn't be opposed, but with my version of systemd, I get this:
```
[lorbus@lenovo os]$ sudo journalctl -b 0 -u NetworkManager --grep="foobarbaz"
-- Logs begin at Mon 2020-09-14 13:11:00 CEST, end at Fri 2021-01-22 21:01:08 CET. --
-- No entries --
[lorbus@lenovo os]$ echo $?
1
[lorbus@lenovo os]$ systemctl --version
systemd 246 (v246.7-2.fc33)
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified

```

Reply:
https://github.com/systemd/systemd/pull/12581

Reply:
(So yeah...we need to handle the old behavior for RHEL8/RHCOS today)